File: MyCodes/Java2/ShapeInheritance/Circle.java

public class Circle extends Shape {
	
	private double radius;
	
	public Circle(double radius) {
		if(radius < 0){
			throw new IllegalArgumentException("Radius must be a positive number.");
		}
		this.radius = radius;
	}
	
	public double getRadius(){
		return radius;
	}
	
	public String getType(){
		return "Circle";
	}
	public double getPerimeter(){
		return (Math.PI * (radius * 2)); 
	}
	
	public double getArea(){
		return (Math.PI * Math.pow(radius, 2)); 
	}
}

File: C++ 2 Submissions/Lab14StevenLake/Lab14/circle.cpp
#include "circle.h"
#include <cmath>
#define _USE_MATH_DEFINES


circle::circle()
    : shape(), radius(0)
{}
    
circle::circle(double r)
    : shape(), radius(r)
{}

circle::circle(double xx, double yy, double r)
    : shape(xx, yy), radius(r)
{}
    
    
double circle::getArea() const{
    return M_PI * radius *radius;
}